### ISSUE TITLE
Table of Contents for Long Pages

### ISSUE THREAD

**Issue by kgudel** [cfpb/hmda-frontend#1487]
Short description explaining the high-level reason for the new issue.

## Current behavior

Long pages such as the FAQs (example: [data-browser FAQ](https://ffiec.cfpb.gov/documentation/2022/data-browser-faq/)) can be difficult to navigate and find the answers a user is looking for.

## Expected behavior

We should include better methods for navigating these pages. One possibility could be a sidebar table of contents as can be seen at fiscaldata.treasury.gov [here](https://fiscaldata.treasury.gov/about-us/#about-fiscal-data)

**Comment by Michaeldremy** [comment]
Update 10.05.2022

Table of contents sidebar has been implemented:
- Users are able to click on different sections and will be automatically navigated to that section
- Old functionality with the developer docs still remain
- Users can send a specific section to another person and when that link is opened up it will take the user directly to the section that was linked

Image to visually show off the table of contents sidebar:

<img width="276" alt="image" src="https://user-images.githubusercontent.com/60801873/194129348-b53d3129-3df7-4951-9782-e0f086909831.png">

Further Updates:
- 2017+ markdown documentation has been standardized #1533 
- Generated new markdown write up which discusses new standardized markdown process. Write-up is commented on this ticket #1565

**Comment by kgudel** [comment]
1. `data-browser-graphs-faq.md`
   1. `How were institutions chosen to be displayed in the quarterly filing graphs?` -> `What institutions are displayed in the Quarterly Graphs`
1. `identifiers-faq.md`
   1. `How can I map 2017 institution identifiers to 2018 institution identifiers?` -> `How can I map 2017 to 2018 institution identifiers?`
   2. `What happens with institution name changes for 2018 and beyond?` -> `What if institution's change their names for 2018 or later`
   3. `What happens with institution name changes for 2017 and prior?` -> `What if institution's change their names for 2017 or prior`
1. `filing-faq`
   1. `What information do I need to provide to register my financial institution?` -> `How do I register my financial institution?`
   2. `What steps should I take on the Co-Applicant Credit Score and Co-Applicant Credit Scoring Model when there is no co-applicant?` -> `What should I record in the Co-Applicant Credit Score and Co-Applicant Credit Scoring Model fields when there is no co-applicant?`
1. `derived-data-fields.md`
   1.  Changing page title to: `Appended HMDA Data Fields`
   2. Adding subheading after the introductory sentences: `Derived HMDA Data Fields`

**Comment by audayshankar** [comment]
Note: My review is still in progress, but starting to add my comments here so I don't post them all at once.

1. [ad-changes.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/ad-changes.md) - Change "New in 2018, derived fields, based on new data points collected, are reported using the CFPB disclosure methodology." to _"New in 2018, derived fields are based on new data points collected and reported using the [CFPB disclosure methodology](https://files.consumerfinance.gov/f/documents/HMDA_Data_Disclosure_Policy_Guidance.Executive_Summary.FINAL.12212018.pdf)."_ (Slight rephrasing and a link added to "CFPB Disclosure methodology")
2. [data-browser-filters.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/data-browser-filters.md) (2018 onward)
    - `Nationwide` Description: Select this option to download the national loan-level dataset. This is a large dataset (5.8 GB) and will take time to download. It will not be able to be opened in standard applications. ADD: _"If using a standard application, adding pre-selected filters will narrow down the dataset with your preferred parameters and minimize the file size."_
    - Some of the filter names aren’t exact matches to what is displayed in the dropdown of the data browser (ex. including/excluding the word derived). The field names listed in the FAQ currently match the field name that‘s in the derived-data-fields.md FAQ but I think they should change to what is currently in the Data Browser. Maybe it could be changed to say something like Age (ageapplicant) so both formats of the field names are displayed? The list of field name differences are below
      - Ageapplicant = Age
      - Derived_ethnicity = Ethnicity
      - Derived_race = Race
      - Derived_sex = Sex
      - Derived_dwelling_category = Dwelling Category
      - Derived_loan_product_type = Loan Product
3. [data-browser-filters.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2017/data-browser-filters.md) (2017) - Data Browser does not currently include 2017 data. Is this prep for a future feature? 
4. [derived-data-fields.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/derived-data-fields.md) - Current: “Researchers and technical professionals interested in the methodology can refer to the [wiki for Derived Fields Categorization](https://github.com/cfpb/hmda-platform/wiki/Derived-Fields-Categorization) or [the Data Disclosure Policy](https://files.consumerfinance.gov/f/documents/HMDA/_Data/_Disclosure/_Policy/_Guidance.Executive/_Summary.FINAL.12212018.pdf).”
    - Data Disclosure Policy link gives me a 404 not found. Here is the correct link: https://files.consumerfinance.gov/f/documents/HMDA_Data_Disclosure_Policy_Guidance.Executive_Summary.FINAL.12212018.pdf
    - The wiki for Derived Fields Categorization leads to a blank page. I will follow-up here on what this link should be.

**Comment by Michaeldremy** [comment]
> 1. `data-browser-graphs-faq.md`
>    
>    1. `How were institutions chosen to be displayed in the quarterly filing graphs?` -> `What institutions are displayed in the Quarterly Graphs`
> 2. `identifiers-faq.md`
>    
>    1. `How can I map 2017 institution identifiers to 2018 institution identifiers?` -> `How can I map 2017 to 2018 institution identifiers?`
>    2. `What happens with institution name changes for 2018 and beyond?` -> `What if institution's change their names for 2018 or later`
>    3. `What happens with institution name changes for 2017 and prior?` -> `What if institution's change their names for 2017 or prior`
> 3. `filing-faq`
>    
>    1. `What information do I need to provide to register my financial institution?` -> `How do I register my financial institution?`
>    2. `What steps should I take on the Co-Applicant Credit Score and Co-Applicant Credit Scoring Model when there is no co-applicant?` -> `What should I record in the Co-Applicant Credit Score and Co-Applicant Credit Scoring Model fields when there is no co-applicant?`
> 4. `derived-data-fields.md`
>    
>    1. Changing page title to: `Appended HMDA Data Fields`
>    2. Adding subheading after the introductory sentences: `Derived HMDA Data Fields`

Changes have been made!

**Comment by Michaeldremy** [comment]
> Note: My review is still in progress, but starting to add my comments here so I don't post them all at once.
> 
> 1. [ad-changes.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/ad-changes.md) - Change "New in 2018, derived fields, based on new data points collected, are reported using the CFPB disclosure methodology." to _"New in 2018, derived fields are based on new data points collected and reported using the [CFPB disclosure methodology](https://files.consumerfinance.gov/f/documents/HMDA_Data_Disclosure_Policy_Guidance.Executive_Summary.FINAL.12212018.pdf)."_ (Slight rephrasing and a link added to "CFPB Disclosure methodology")
> 2. [data-browser-filters.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/data-browser-filters.md) (2018 onward)
>    
>    * `Nationwide` Description: Select this option to download the national loan-level dataset. This is a large dataset (5.8 GB) and will take time to download. It will not be able to be opened in standard applications. ADD: _"If using a standard application, adding pre-selected filters will narrow down the dataset with your preferred parameters and minimize the file size."_
> 3. [data-browser-filters.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2017/data-browser-filters.md) (2017) - Data Browser does not currently include 2017 data. Is this prep for a future feature? This file is not needed at this time.
> 4. [derived-data-fields.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/derived-data-fields.md) - Current: “Researchers and technical professionals interested in the methodology can refer to the [wiki for Derived Fields Categorization](https://github.com/cfpb/hmda-platform/wiki/Derived-Fields-Categorization) or [the Data Disclosure Policy](https://files.consumerfinance.gov/f/documents/HMDA/_Data/_Disclosure/_Policy/_Guidance.Executive/_Summary.FINAL.12212018.pdf).”
>    
>    * Data Disclosure Policy link gives me a 404 not found. Here is the correct link: https://files.consumerfinance.gov/f/documents/HMDA_Data_Disclosure_Policy_Guidance.Executive_Summary.FINAL.12212018.pdf
>    * The wiki for Derived Fields Categorization leads to a blank page. I will follow-up here on what this link should be. @audayshankar here's the link to that: https://github.com/cfpb/hmda-platform/wiki/Derived-Fields-Categorization-2018-Onward
> 5. [data-collection-timelines.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/data-collection-timelines.md) - Current: "PLEASE NOTE: Collection of 2018 HMDA data ends on Dec 31, 2021." Change to _PLEASE NOTE: Collection of 2019 HMDA data ends on Dec 31, 2022._ Maybe for just 2019 onward documents?

Image will be not show up until branch has been merged into master.

> 6. [file-format-verification.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2019/file-format-verification.md) – Remove step one because you no longer have the option to select an applicable year. Step 1 could be changed to “1. Navigate to the File Format Verification Tool at [https://ffiec.cfpb.gov/tools/file-format-verification”](https://ffiec.cfpb.gov/tools/file-format-verification%E2%80%9D). Screenshot needs to be replaced to the below:
>    ![image](https://user-images.githubusercontent.com/58990254/197864983-a835e796-f549-41ba-90d7-a6b85fea3dd6.png)
> 7. [filing-faq.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/filing-faq.md)
>    
>    * At end of the section `What tools can I use to help prepare my file for submission?` - the following sentence needs to be removed “Additionally, there is a unique File Format Verification Tool for each HMDA data collection year, so please select the relevant year before uploading a file.”
>    * Section `When can I submit my file?` has the following sentence through out all years “Annual HMDA data submissions for the 2019 filing year are accepted January 1 - March 1, 2020.” To make it more standardized/applicable to all years, it could be adjusted to say Annual HMDA data submissions for the filing year are accepted between January 1 – March to be considered timely”
> 8. [lar-data-fields.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/lar-data-fields.md) - `[debt_to_income_ratio](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/lar-data-fields.md#debt_to_income_ratio)` - bullet point for 60% seems to be formatted oddly compared to the rest

Above changes have been made.

Still WIP:

> 2. [data-browser-filters.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/data-browser-filters.md) (2018 onward)
>
>    * Some of the filter names aren’t exact matches to what is displayed in the dropdown of the data browser (ex. including/excluding the word derived). The field names listed in the FAQ currently match the field name that‘s in the derived-data-fields.md FAQ but I think they should change to what is currently in the Data Browser. Maybe it could be changed to say something like Age (ageapplicant) so both formats of the field names are displayed? The list of field name differences are below
>      
>      * Ageapplicant = Age
>      * Derived_ethnicity = Ethnicity
>      * Derived_race = Race
>      * Derived_sex = Sex
>      * Derived_dwelling_category = Dwelling Category
>      * Derived_loan_product_type = Loan Product

Waiting on further clarification.

>9.  [modified-lar-schema.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/modified-lar-schema.md) – Table formatting seems to have gotten messed up

Waiting on an update for this one.

> 10. Get teams thoughts on sneaking an OLARFT link into [lar-formatting.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2021/lar-formatting.md)

**Comment by Michaeldremy** [comment]
> 1. [data-browser-faq.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/data-browser-faq.md) - This FAQ is for the HMDA Dataset Filtering Tool, not the HMDA Data Browser as a whole (since the Data Browser now has Maps & Graphs).
>    Recommend changing any instance of HMDA Data Browser on this page to HMDA Dataset Filtering and changing the answer to the last question "Which dataset is used by the Data Browser?" to "The HMDA Dataset Filtering tool uses the latest available [static dataset](https://ffiec.cfpb.gov/data-publication/) (Three Year, One Year, Snapshot) for the selected year."
> 2. [data-browser-filters.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/data-browser-filters.md) (2018 onward)
>    Recommend changing HMDA Data Browser to HMDA Dataset Filtering Tool
> 3. [file-format-verification.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/file-format-verification.md)
> 
> * There should be a new screenshot for the FFVT instruction # 1 as 2017 and 2018 are no longer files that can be submitted.
> * Recommend updating the FAQs to the latest link https://ffiec.cfpb.gov/documentation/2022/faqs/
> 
> 4. [filing-faq.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/filing-faq.md)
> 
> * Current: In 'Filing Preparation' Section, under https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/filing-faq.md#what-tools-can-i-use-to-help-prepare-my-file-for-submission: "You can use the [Filing Instructions Guide (FIG)](https://s3.amazonaws.com/cfpb-hmda-public/prod/help/2018-hmda-fig.pdf) applicable to the year you are filing data for. Filing Instructions Guides for each year can be found on the home page of https://ffiec.cfpb.gov/, under the “Help For Filers” section." to "You can use the [Filing Instructions Guide (FIG)](https://s3.amazonaws.com/cfpb-hmda-public/prod/help/2018-hmda-fig.pdf) applicable to the year you are filing data for. Filing Instructions Guides for each year can be found on the home page of https://ffiec.cfpb.gov/, **under the “Guides For HMDA Filers” section.**
> * Change: In 'Submission' Section, under "When can I submit my file?" Collection of 2018 HMDA data ended on Dec 31, 2021.
> 
> 5. [static-dataset-faq.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/static-dataset-faq.md) -
> 
> * Change: For 'Snapshot' row listed in the table, change "1 month after the timely filing deadline "to "Typically 1 month" - so that we are not held to April 1 every year.
> * For "Which dataset is used by the Data Browser?" Add a link to the [Graphs](https://ffiec.cfpb.gov/data-browser/graphs/quarterly/applications?periodLow=2019-Q1&periodHigh=2022-Q2&visibleSeries=Conventional%20Conforming,Conventional%20Non-Conforming,FHA,HELOC,RHS/FSA,VA) in the "The HMDA Data Browser tools" sentence.

Above changes have been made.

Waiting on clarification.

> 4. [filing-faq.md](https://github.com/cfpb/hmda-frontend/blob/1487-table-of-contents/src/documentation/markdown/2018/filing-faq.md)
> 
> * Change: In 'Submission' Section, under "When can I submit my file?" Collection of 2018 HMDA data ended on Dec 31, 2021.
